File: client/web/sudoo/lib/domain/repository/auth_repository.dart
import 'package:sudoo/domain/core/data_state.dart';
import 'package:sudoo/domain/model/auth/account.dart';
import 'package:sudoo/domain/model/auth/auth_config.dart';
import 'package:sudoo/domain/model/auth/token.dart';
import 'package:sudoo/domain/model/auth/verify_otp.dart';

abstract class AuthRepository {
  Future<DataState<AuthConfig, Exception>> getConfig();

  Future<DataState<Token, Exception>> signIn(Account account);

  Future<DataState<dynamic, Exception>> signUp(Account account);

  Future<DataState<Token, Exception>> refreshToken();

  Future<DataState<Token, Exception>> submitOtp(VerifyOtp verifyOtp);

  Future<DataState<dynamic, Exception>> changePassword(
      String oldPassword, String newPassword);

  Future<DataState<dynamic, Exception>> logout();
}

File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/domain/repository/AuthRepository.kt
package com.sudo248.sudoo.domain.repository

import com.sudo248.base_android.core.DataState
import com.sudo248.sudoo.domain.entity.auth.Account
import com.sudo248.sudoo.domain.entity.auth.AuthConfig
import com.sudo248.sudoo.domain.entity.auth.Token


/**
 * **Created by**
 *
 * @author *Sudo248*
 * @since 23:47 - 04/03/2023
 */
interface AuthRepository {
    suspend fun getAuthConfig(): DataState<AuthConfig, Exception>
    suspend fun refreshToken(): DataState<Token, Exception>
    suspend fun saveToken(token: Token): DataState<Unit, Exception>
    suspend fun signIn(account: Account): DataState<Token, Exception>
    suspend fun signUp(account: Account): DataState<Unit, Exception>
    suspend fun generateOtp(phoneNumber: String): DataState<Unit, Exception>
    suspend fun verifyOtp(phoneNumber: String, otp: String): DataState<Token, Exception>
    suspend fun changePassword(oldPassword: String, newPassWord: String): DataState<Unit, Exception>
    suspend fun logout(): DataState<Unit, Exception>
}